tehom/downloads.py

```diff
@@ -536,32 +536,69 @@ def _interpolate_and_group_ais(
     """Interpolate the lat/lon of ships to the specified time.
 
     Interpolation rules:
-        A ship has observations near the specified time, before and
+        C1: A ship has observations near the specified time, before and
             after: linear interpolation
-        A ship has one observation very near the specified time, either
+        C2: A ship has one observation very near the specified time, either
             before or after, but not both: constant interpolation
-        A ship does not meet above criteria: do not create an
+        C3: A ship does not meet above criteria: do not create an
             interpolated record for this ship at this time
 
     Note:
         What counts as "near" and "very near" is subject to change and
         may be refactored out into an interpolation parameters object
 
+        MWM: "near" = entries before and after the given time,
+             "very near" = one entry within a neighborhood of the given time
+
     Arguments:
         ais_df: ship records, including a basedatetime column.
         times: when to interpolate the ship positions.
 
     Returns:
         The interpolated records, grouped by time.
     """
-    # Morgan, you'll have to first groupby mmsi (ship unique id) and
+    # 1.) groupby mmsi (ship unique id)
     # then apply an interpolation function for each timepoint.  The
     # interpolation function will take a dataframe and a timepoint,
     # and will determine, based on the nearest records before/after
     # the timepoint, which interpolation rule to apply.
     #
     # While this function sounds like it takes a long time, its ok at
     # the outset to accomplish this somewhat inefficiently.
+    mmsi_set = ais_df.groupby["mmsi"]
```

Comment:
groupby is a function, needs parentheses (e.g. `groupby("mmsi")`)

Also, be careful naming things `*_set` when they're not a set object.  Try `mmsi_gb` since this is a groupby object.